src/internal.c

```diff
@@ -9405,18 +9379,25 @@ static void DoCertFatalAlert(WOLFSSL* ssl, int ret)
  * store->error_depth member to determine index (0=peer, >1 intermediates)
  */
 
-static int DoVerifyCallback(WOLFSSL* ssl, int ret, ProcPeerCertArgs* args)
+int DoVerifyCallback(WOLFSSL_CERT_MANAGER* cm, WOLFSSL* ssl, int ret,
+                                                        ProcPeerCertArgs* args)
 {
     int verify_ok = 0, use_cb = 0;
-
+#ifdef WOLFSSL_SMALL_STACK
```

Comment:
Let's make the "heap" always available and change all XMALLOC/XFREE in this function to use "heap". I see a mix of `ssl->heap` and `heap`... might lead to future issues.